src/backend/Storage.cpp

```diff
@@ -136,18 +136,21 @@ namespace cytnx {
 
     unsigned int IDDs = 999;
     f.write((char *)&IDDs, sizeof(unsigned int));
-    f.write((char *)&this->size(), sizeof(unsigned long long));
-    f.write((char *)&this->dtype(), sizeof(unsigned int));
-    f.write((char *)&this->device(), sizeof(int));
+    auto write_number = [&f](auto number) {
+      f.write(reinterpret_cast<char *>(&number), sizeof(number));
+    };
+    write_number(this->size());
+    write_number(this->dtype());
+    write_number(this->device());
```

Comment:
`this->size()` and the other statements are not `lvalue` now, so we cannot access the address by `&`.